### DISCUSSION TITLE
flameshot: not sandboxed when started by KDE Plasma keyboard shortcuts (dbus)

### DISCUSSION THREAD

**Discussion by tredondo** [netblue30/firejail#4092] [Q&A]
Flameshot has a firejail profile, but how can it be used when launching Flameshot in KDE Plasma via shortcut keys, as described in [their guide](https://github.com/flameshot-org/flameshot#on-kde-plasma-desktop)? That file creates a KDE Menu entry like this:

![image](https://user-images.githubusercontent.com/10043369/111026423-428c0780-838e-11eb-83fc-9d75fc2cd22a.png)

Relates to:

* #3539

**Comment by ghost** [comment]
The [flameshot profile](https://github.com/netblue30/firejail/blob/master/etc/profile-a-l/flameshot.profile) uses dbus-user filter and dbus-user.own `org.flameshot.Flameshot` is already there.

That being said, I'm not familiar with KDE, so not sure if they use hardcoded paths to /usr/bin/flameshot (or anything else involved) that might break firejail integration. I would go over all the items in the section your screenshot shows, check if there are other D-Bus addresses besides org.flameshot.Flameshot being used and add those to a flameshot.local file to test. It might need adding stuff to `private-bin` too depending on what applications are called by these shortcuts.

**Comment by rusty-snake** [comment]
[firecfg.py](https://github.com/rusty-snake/firecfg.py) can firejail D-Bus activated programs (but only on the session bus for now). However, flameshot crashes if you do so.
